test/run_test.py

```diff
@@ -0,0 +1,390 @@
+        # clients info recorded in the backup file
+        try:
+            proc_server.wait(5)
+        except subprocess.TimeoutExpired:
+            proc_server.kill()
+
+        # Relaunch the server again and expect it reloads the backup data.
+        result = subprocess.run([discovery_server_tool_path, path])
+
+        if result.returncode:
+            logger.error(f'Failure while running the backup server for {test}')
+            logger.error(result.stdout)
+            logger.error(result.stderr)
+            exit(result.returncode)
+
+        # Wait for client completion
+        proc_client.communicate()
+
+        if proc_client.returncode:
+            logger.error(f'Failure when running clients for {test}')
+            logger.error(result.stderr)
+            exit(proc_client.returncode)
+
+    elif test == 'test_26_backup_restore':
+        aux_test_path = os.path.join(os.path.dirname(path), f'{test}_1.xml')
+        # Launch and wait for completion
+        proc_main = subprocess.run(
+            [discovery_server_tool_path, path])
+        if proc_main.returncode:
+            logger.error(
+                f'{test} process ha failed to launch the backup server: '
+                f'returncode {proc_main.returncode}')
+            exit(proc_main.returncode)
+
+        proc_restore = subprocess.run(
+            [discovery_server_tool_path, aux_test_path])
+
+        if proc_restore.returncode:
+            logger.error(
+                f'{test} process has failed to restore the backup server: '
+                f'returncode {proc_restore.returncode}')
+            exit(proc_restore.returncode)
+
+    else:
+        subprocess.run([discovery_server_tool_path, path])
+
+
+def validate_test(
+    test_params_df,
+    test_path,
+    test_snapshot,
+    ground_truth_snapshot,
+    discovery_server_tool_path,
+    fds_path=None,
+    debug=False
+):
+    """
+    Execute the tests and validate the output.
+
+    :param test_params_df: The test parameters in a pandas Dataframe format.
+    :param test_path: The path of the xml configuration file of the test.
+    :param test_snapshot: Path to the snapshot xml file containing the
+            Discovery-Server test output.
+    :param ground_truth_snapshot: The path to the snapshot xml file containing
+        the Discovery-Server ground-truth test output.
+    :param discovery_server_tool: The path to the discovery server executable.
+    :param fds_path: The path to the fast-discovery-server tool.
+    :param debug: Debug flag (Default: False).
+    """
+    test = test_params_df.iloc[0]['test_name']
+    logger.info('------------------------------------------------------------')
+    logger.info(f'Running {test}')
+    execute_test(
+        test, test_path, discovery_server_tool_path, fds_path, debug)
+    logger.info('------------------------------------------------------------')
+
+    validation_result = True
+
+    validators = [
+        clv.CountLinesValidation,
```

Comment:
It also needs a way to check if a stderr has been thrown during the execution.
* This would be tricki, because if you redirect the stderr output in `subprocess` it is not shown in the log. A possibility would be command `tee`, but windows does not have it :+1: . Another option would be to have a separate thread reading and printing the output.
Good luck.